psydac/api/tests/test_api_expr_2d_scalar.py

```diff
@@ -18,7 +18,9 @@
 def run_poisson_2d_dir(ncells, degree, comm=None):
 
     # ... abstract model
-    domain = Square()
+    domain  = Square()
+    mapping = IdentityMapping('M',2)
+    domain  = mapping(domain)
```

Comment:
What is the point of this change?
The only unit test in this file is marked as `xfail`

Reply:
I don't remember anymore, but I think it's because we need a mapping otherwise it will give us an error

Reply:
Why? In general we can handle domains with no mapping. If `discretize(expr)` does not work in such a case we should fix it.

Reply:
yes, we should be able to handle expressions with no mapping, it's just that I didn't have the time to debug it and fix the issue

Reply:
OK, let's leave this for another PR. Please create a new issue!

Reply:
done https://github.com/pyccel/psydac/issues/177